### PR TITLE
Add git to list of deployed modules

### PR DESCRIPTION
**Pull request by wizmer**

### CODE REVIEW

File: deploy/configs/applications/modules.yaml [hunk 1]
```diff
@@ -37,6 +37,7 @@ modules:
       - connectome-tools
       - emsim
       - functionalizer
+      - git
```
Comment (matz-e) [inline deploy/configs/applications/modules.yaml]: This does not go here, since we don't develop git.

**Review by matz-e — CHANGES_REQUESTED** [review]
Should be moved into the `tools` config, that's where it's deployed, too.

**Comment by pramodk** [comment]
I am sure the answer is no but just to ask - I assume we are not using any new git features that Spack will need while cloning repo? (e.g. for sibmodules).

**Comment by wizmer** [comment]
Just to try to understand the point of your question. The module would not be loaded except if explicitly required, no ? How can this affect the Spack workflow ?

**Comment by matz-e** [comment]
> I am sure the answer is no but just to ask - I assume we are not using any new git features that Spack will need while cloning repo? (e.g. for sibmodules).

I'm not sure why a newer git is essential (although I have a private module), but I think things like `--force-with-lease` are not in the stock version. @wizmer is that the feature you desire?

**Comment by wizmer** [comment]
Yes, --force-with-lease and also the new way of managing branches. It seems to me a bit faster as well.

**Comment by matz-e** [comment]
@pramodk while we want to minimize the modules present, I think this is a scenario where it would make sense to enable `git`. After all, our own team has endorsed functionality present only in newer `git`s…

**Comment by matz-e** [comment]
To add, contrary to `emacs` :angry:, this would probably be a wider used module

**Review by matz-e — DISMISSED** [review]

**Review by matz-e — APPROVED** [review]

**Pull request merged**